File: programers_Lv1/크기가 작은 부분 문자열.py
# https://school.programmers.co.kr/learn/courses/30/lessons/147355?language=python3
def solution(t, p):
    answer = 0
    slice_length = len(p)
    i = 0
    while True:
        temp = t[i : slice_length + i]
        if len(temp) < slice_length:
            break
        if int(temp) <= int(p):
            answer += 1
        i += 1

    return answer


# print(solution("3141592", "271"))  # 2
print(solution("500220839878", "7"))  # 8

File: programers_Lv1/크기가 작은 부분 문자열.js
// https://school.programmers.co.kr/learn/courses/30/lessons/147355?language=javascript
function solution(t, p) {
  var answer = 0;
  const numLength = p.length;
  let i = 0;

  while (1) {
    num = t.slice(i, numLength + i);
    if (num.length !== numLength) {
      break;
    }
    answer += +num <= +p ? 1 : 0;
    i++;
  }

  return answer;
}

console.log(solution("500220839878", "7")); // 8
